File: go/internal/replay/parserfixture/testdata/trees/haskell/OrderService.hs
module Example.OrderService
  ( placeOrder
  , findOrder
  ) where

import Example.OrderRepository (Order, byId, createOrder, saveOrder)

placeOrder :: String -> [String] -> Order
placeOrder customerId skus =
  saveOrder (foldr addLine (createOrder customerId) skus)

addLine :: String -> Order -> Order
addLine sku order = order

findOrder :: String -> Maybe Order
findOrder = byId

File: go/internal/replay/parserfixture/testdata/trees/perl/order_service.pl
#!/usr/bin/perl
use strict;
use warnings;

sub place_order {
    my ($repository, $customer_id, $skus) = @_;
    my $order = $repository->create($customer_id);
    for my $sku (@{$skus}) {
        add_line($order, $sku);
    }
    return $repository->save($order);
}

sub add_line {
    my ($order, $sku) = @_;
    push @{ $order->{lines} }, $sku;
    return $order;
}

sub find_order {
    my ($repository, $order_id) = @_;
    return $repository->by_id($order_id);
}

1;
